src/foam/mlang/mlang.js

```diff
@@ -339,6 +339,10 @@ foam.CLASS({
     {
       name: 'adapt',
       value: function(_, o) {
+        if ( foam.Object.isInstance(o) && o.class ) {
+          var obj = (foam.lookup(o.class))?.create(o);
```

Comment:
Doesn't set the sub-context in the create.
When was this code changed, since the normal adapt supports this:
    {
      name: 'adapt',
      value: function(_, v, prop) {
        // All FObjects may be null.
        if ( v === null ) return v;

        var type = foam.lookup(prop.type);

        return type.isInstance(v) ?
          v :
          ( v.class ?
            this.__context__.lookup(v.class) :
            type ).create(v, this.__subContext__);
      }
    },

It would be better to first check if the type is a function and if not, call the old function.
Like:
value: (function() {
  const OLD_ADAPT = foam.core.FObjectProperty.ADAPT.value;;
  return function(_, o, prop) {
    if ( foam.Function.isInstance(o) && ! o.f ) return ...;
    return OLD_ADAPT(_, o, prop);
  };
 })()